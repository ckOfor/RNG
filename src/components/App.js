// react
import React, { Component } from 'react';

// third party
import {
  Navbar,
  NavItem,
  Button,
  Row,
  Input,
  Dropdown,
  Modal,
  Table,
} from 'react-materialize'

// image
import logo from '../logo.svg';


// styles
import '../App.css';

// const fs = require('browserify-fs');
import fs from 'fs';

class App extends Component {
  state={
    generatedNumbers: [],
    userRange: '',
    minValue: '',
    maxValue: '',
    selected: '',
    mounted: false
  }
  
  componentDidMount(){
    this.mounted = true;
  }
  
  componentWillMount() {
    fs.readFile('../../data.txt', 'utf-8', (err, generatedNumbers) => !err
      ? this.setState({generatedNumbers: JSON.parse(generatedNumbers) }) : null); // eslint-disable-line
  }
  
  componentWillUnmount(){
    this.mounted = false;
  }
  
  onGenerateButtonClick = (userRange) => {
    return userRange > 0 && userRange <= 5000 &&
      this.phoneNumberGenerator(userRange)
  }
  
  phoneNumberGenerator = (userRange) => {
    let phoneNumbers, generatedNumbers = [];
    
    for ( let i = 0; i < userRange; i++) {
      phoneNumbers = `0${Math.floor(Math.random() * 1111111111)}`
      generatedNumbers.push({ id: i + 1,  value: phoneNumbers });
    }
    
    this.setState({ generatedNumbers })
    const dataToWrite = JSON.stringify(generatedNumbers);
    fs.writeFile('../../numbers.txt', dataToWrite, () => {});
    
    this.returnMaxAndMin(generatedNumbers)
  }
  
  sortGeneratedNumbers = (generatedNumbers, choice) => {
    if(choice === "asc") {
      this.setState({
        generatedNumbers: generatedNumbers.sort( function ( a, b )
        { return a.value - b.value; })
      })
      
      return generatedNumbers
    }
    
    if(choice === "dsc") {
      this.setState({
        generatedNumbers: generatedNumbers.sort( function ( a, b )
        { return b.value - a.value; })
      })
      
      return generatedNumbers
    }
  }
  
  returnMaxAndMin = (generatedNumbers, whatToReturn) => {
    let minValue = generatedNumbers[0].value;
    let maxValue = 0;
    
    Object.keys(generatedNumbers).forEach((value) => {
      if (generatedNumbers[value].value < minValue) {
        minValue = generatedNumbers[value].value
      }
    })
    
    Object.keys(generatedNumbers).forEach((value) => {
      if (maxValue < generatedNumbers[value].value) {
        maxValue = generatedNumbers[value].value
      }
    })
    
    this.setState({ minValue, maxValue })
  }
  
  render() {
    let { generatedNumbers } = this.state
    let num = [1,2,3,4,5]
    return (
      <div className="App">
        <Navbar brand='RNG' right>
          <NavItem>
            Total : { Object.keys(this.state.generatedNumbers).length && Object.keys(this.state.generatedNumbers).length }
          </NavItem>
        </Navbar>
        
        {/*<TableComponent list={this.state.generatedNumbers} />*/}
        <div style={{ marginLeft: 300, height: 500, width: 1000, marginTop: 50, display: 'block',
          overflow:'auto' }}>
          <Table className="table">
            <thead>
            <tr>
              <th data-field="id">S/N</th>
              <th data-field="name">Unique ID</th>
              <th data-field="price">Phone Numbers</th>
            </tr>
            </thead>
            <tbody>
            {
              (
                Object.keys(generatedNumbers).map((row, index) => {
                  return (
                    <tr key={index}>
                      <td>{index + 1}</td>
                      <td>{generatedNumbers[row].id}</td>
                      <td>{generatedNumbers[row].value}</td>
                    </tr>
                  )
                })
              )
            }
            </tbody>
          </Table>
        </div>
        <div style={{ marginLeft: 300, marginTop: 50 }}>
          <Row>
            {/*<Dropdown trigger={<Button style={{ marginTop: 20, marginLeft: 40  }}>Filter by:</Button>}>*/}
            {/*<NavItem*/}
            {/*className="asc"*/}
            {/*onClick={() => this.sortGeneratedNumbers(this.state.generatedNumbers, "asc")}>Ascending</NavItem>*/}
            {/*<NavItem*/}
            {/*className="dsc"*/}
            {/*onClick={() => this.sortGeneratedNumbers(this.state.generatedNumbers, "dsc")}>Descending</NavItem>*/}
            {/*<NavItem divider />*/}
            {/*<Modal*/}
            {/*header={`Minimum: ${this.state.minValue}`}*/}
            {/*trigger={<NavItem className="min">Min</NavItem>}>*/}
            {/*</Modal>*/}
            {/*<Modal*/}
            {/*header={`Maximum: ${this.state.maxValue}`}*/}
            {/*trigger={<NavItem className="max">Max</NavItem>}>*/}
            {/*</Modal>*/}
            {/*</Dropdown>*/}
            
            <Input
              className="range"
              s={2}
              label="Enter Range 1 <= 5000"
              onChange={userRange => this.setState({ userRange: userRange.target.value.replace(/\D/g, '') })}
              value={this.state.userRange}
            />
            <Button
              className="generate"
              style={{ marginTop: 20, marginLeft: 40 }}
              s={6}
              onClick={() => this.onGenerateButtonClick(this.state.userRange)}
              waves='light'>
              GENERATE
            </Button>
            
            <Button
              className="filter"
              style={{ marginTop: 20, marginLeft: 40 }}
              s={6}
              // onClick={() => this.onGenerateButtonClick(this.state.userRange)}
              waves='light'>
              filter
            </Button>
          </Row>
        </div>
      </div>
    );
  }
}

export default App;
