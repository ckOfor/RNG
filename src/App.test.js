// react
import React from 'react';

// third-party libraries
import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json';
import { spy, sinon } from 'sinon';

// component
import App from './components/App';

// const mock = requiare('mock-fs');

import mock from 'mock-fs'

let event = {
  target: {
    name: 'numberToGenerate',
    value: 1000,
    selectedOptions: [
      { innerHTML: 'selected name' },
    ],
  },
  preventDefault: spy(),
};

const mockData = [
  {
    id: 1,
    value: '0426788013',
  },
  {
    id: 2,
    value: '0159929077',
  },
];


describe('App', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = mount(<App />);
  });
  
  afterAll(() => {
    mock.restore();
    wrapper.unmount();
  });
  
  it('should ensure ascending order is clicked', () => {
    const wrapper = mount(<App />);
    const action = wrapper.find('NavItem.asc').simulate('click');
    expect(action).toBeInstanceOf(Object);
    wrapper.unmount();
  });
  
  
  it('should ensure descending order is clicked', () => {
    const wrapper = shallow(<App />);
    const action = wrapper.find('NavItem.dsc').simulate('click');
    console.log(action)
  });
  
  it('should ensure asc order is clicked', () => {
    const wrapper = shallow(<App />);
    const action = wrapper.find('NavItem.asc').simulate('click');
    console.log(action)
  });
  
  it('should ensure dropdown is clicked', () => {
    const wrapper = mount(<App />);
    const action = wrapper.find('Dropdown.dropdown').simulate('click');
    expect(action).toBeInstanceOf(Object);
    wrapper.unmount();
  });

  it('should call the onGenerateButtonClick method', () => {
    const wrapper = mount(<App />);
    const onGenerateButtonClickSpy = spy(wrapper.instance(), 'onGenerateButtonClick');
    wrapper.instance().onGenerateButtonClick(100);
    expect(onGenerateButtonClickSpy.called).toEqual(true);
    wrapper.unmount();
  });
  
  it('should call the updateRange method ', () => {
    const wrapper = mount(<App />);
    const updateRangeSpy = spy(wrapper.instance(), 'updateRange');
    wrapper.instance().updateRange(event);
    expect(updateRangeSpy.called).toEqual(true);
    wrapper.unmount();
  });

  it('should call the sortGeneratedNumbers method using asc ', () => {
    const wrapper = mount(<App />);
    const sortGeneratedNumbersSpy = spy(wrapper.instance(), 'sortGeneratedNumbers');
    wrapper.instance().sortGeneratedNumbers(
      [
        { id: 1, value: "0911276029" },
        { id: 2, value: "0780997650" },
        { id: 3, value: "0141172042" },
      ],
      "asc"
    );
    expect(sortGeneratedNumbersSpy.called).toEqual(true);
    wrapper.unmount();
  });
  
  it('should call the phoneNumberGenerator method using range of 10 ', () => {
    const wrapper = mount(<App />);
    const phoneNumberGeneratorSpy = spy(wrapper.instance(), 'phoneNumberGenerator');
    wrapper.instance().phoneNumberGenerator(10);
    expect(phoneNumberGeneratorSpy.called).toEqual(true);
    wrapper.unmount();
  });
  

  it('should call the sortGeneratedNumbers method using dsc ', () => {
    const wrapper = mount(<App />);
    const sortGeneratedNumbersSpy = spy(wrapper.instance(), 'sortGeneratedNumbers');
    wrapper.instance().sortGeneratedNumbers(
      [
        { id: 1, value: "0911276029" },
        { id: 2, value: "0780997650" },
        { id: 3, value: "0141172042" },
      ],
      "dsc"
    );
    expect(sortGeneratedNumbersSpy.called).toEqual(true);
    wrapper.unmount()
  });
  
  it('renders without crashing', () => {
    const rendering = toJson(wrapper);
    expect(rendering).toMatchSnapshot();
  });
  
  it('should call the setComponentState method', () => {
    const wrapper = mount(<App />);
    const setComponentStateSpy = spy(wrapper.instance(), 'setComponentState');
    wrapper.instance().setComponentState(mockData);
    expect(setComponentStateSpy.called).toEqual(true);
    expect(wrapper.state().generatedNumbers).toEqual(mockData);
    // expect(wrapper.state().numberUpdated).toEqual(true);
    wrapper.unmount()
  });
  
  
});
