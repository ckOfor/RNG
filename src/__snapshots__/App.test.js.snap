// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<App>
  <div
    className="App"
  >
    <Navbar
      brand="RNG"
      href="/"
      options={Object {}}
      right={true}
    >
      <nav>
        <div
          className="nav-wrapper"
        >
          <Col
            s={12}
          >
            <div
              className="col s12"
            >
              <a
                className="brand-logo"
                href="/"
              >
                RNG
              </a>
              <ul
                className="right hide-on-med-and-down"
              >
                <NavItem>
                  <li>
                    <a
                      href=""
                    >
                      Total : 
                      0
                    </a>
                  </li>
                </NavItem>
              </ul>
              <ul
                className="side-nav"
                id="nav-mobile"
              >
                <NavItem>
                  <li>
                    <a
                      href=""
                    >
                      Total : 
                      0
                    </a>
                  </li>
                </NavItem>
              </ul>
              <a
                className="button-collapse"
                data-activates="nav-mobile"
                href="#"
              >
                <Icon>
                  <i
                    className="material-icons"
                  >
                    view_headline
                  </i>
                </Icon>
              </a>
            </div>
          </Col>
        </div>
      </nav>
    </Navbar>
    <div
      style={
        Object {
          "display": "block",
          "height": 500,
          "marginLeft": 300,
          "marginTop": 50,
          "overflow": "auto",
          "width": 1000,
        }
      }
    >
      <Table
        className="table"
      >
        <table
          className="table"
        >
          <thead>
            <tr>
              <th
                data-field="id"
              >
                S/N
              </th>
              <th
                data-field="name"
              >
                Unique ID
              </th>
              <th
                data-field="price"
              >
                Phone Numbers
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </Table>
    </div>
    <div
      style={
        Object {
          "marginLeft": 300,
          "marginTop": 50,
        }
      }
    >
      <Row>
        <div
          className="row"
        >
          <Dropdown
            className="dropdown"
            onClick={[Function]}
            trigger={
              <Button
                node="button"
                style={
                  Object {
                    "marginLeft": 40,
                    "marginTop": 20,
                  }
                }
              >
                Filter by:
              </Button>
            }
          >
            <span>
              <Button
                className="dropdown-button"
                data-activates="dropdown_64"
                node="button"
                style={
                  Object {
                    "marginLeft": 40,
                    "marginTop": 20,
                  }
                }
              >
                <button
                  className="btn dropdown-button"
                  data-activates="dropdown_64"
                  disabled={false}
                  style={
                    Object {
                      "marginLeft": 40,
                      "marginTop": 20,
                    }
                  }
                >
                  Filter by:
                </button>
              </Button>
              <ul
                className="dropdown-content dropdown"
                id="dropdown_64"
                onClick={[Function]}
              >
                <NavItem
                  className="asc"
                  onClick={[Function]}
                >
                  <li
                    className="asc"
                  >
                    <a
                      onClick={[Function]}
                    >
                      Ascending
                    </a>
                  </li>
                </NavItem>
                <NavItem
                  className="dsc"
                  onClick={[Function]}
                >
                  <li
                    className="dsc"
                  >
                    <a
                      onClick={[Function]}
                    >
                      Descending
                    </a>
                  </li>
                </NavItem>
                <NavItem
                  divider={true}
                >
                  <li
                    className="divider"
                  />
                </NavItem>
                <Modal
                  actions={
                    Array [
                      <Button
                        flat={true}
                        modal="close"
                        node="button"
                        waves="light"
                      >
                        Close
                      </Button>,
                    ]
                  }
                  bottomSheet={false}
                  fixedFooter={false}
                  header="Minimum: "
                  modalOptions={Object {}}
                  trigger={
                    <NavItem
                      className="min"
                    >
                      Min
                    </NavItem>
                  }
                >
                  <div>
                    <NavItem
                      className="min"
                      onClick={[Function]}
                    >
                      <li
                        className="min"
                      >
                        <a
                          onClick={[Function]}
                        >
                          Min
                        </a>
                      </li>
                    </NavItem>
                    <Portal
                      containerInfo={
                        <div>
                          <div
                            class="modal"
                            id="modal_65"
                          >
                            <div
                              class="modal-content"
                            >
                              <h4>
                                Minimum: 
                              </h4>
                            </div>
                            <div
                              class="modal-footer"
                            >
                              <button
                                class="btn waves-effect waves-light btn-flat modal-action modal-close"
                              >
                                Close
                              </button>
                            </div>
                          </div>
                        </div>
                      }
                    >
                      <div
                        className="modal"
                        id="modal_65"
                      >
                        <div
                          className="modal-content"
                        >
                          <h4>
                            Minimum: 
                          </h4>
                        </div>
                        <div
                          className="modal-footer"
                        >
                          <Button
                            flat={true}
                            key=".0"
                            modal="close"
                            node="button"
                            waves="light"
                          >
                            <button
                              className="btn waves-effect waves-light btn-flat modal-action modal-close"
                              disabled={false}
                            >
                              Close
                            </button>
                          </Button>
                        </div>
                      </div>
                    </Portal>
                  </div>
                </Modal>
                <Modal
                  actions={
                    Array [
                      <Button
                        flat={true}
                        modal="close"
                        node="button"
                        waves="light"
                      >
                        Close
                      </Button>,
                    ]
                  }
                  bottomSheet={false}
                  fixedFooter={false}
                  header="Maximum: "
                  modalOptions={Object {}}
                  trigger={
                    <NavItem
                      className="max"
                    >
                      Max
                    </NavItem>
                  }
                >
                  <div>
                    <NavItem
                      className="max"
                      onClick={[Function]}
                    >
                      <li
                        className="max"
                      >
                        <a
                          onClick={[Function]}
                        >
                          Max
                        </a>
                      </li>
                    </NavItem>
                    <Portal
                      containerInfo={
                        <div>
                          <div
                            class="modal"
                            id="modal_66"
                          >
                            <div
                              class="modal-content"
                            >
                              <h4>
                                Maximum: 
                              </h4>
                            </div>
                            <div
                              class="modal-footer"
                            >
                              <button
                                class="btn waves-effect waves-light btn-flat modal-action modal-close"
                              >
                                Close
                              </button>
                            </div>
                          </div>
                        </div>
                      }
                    >
                      <div
                        className="modal"
                        id="modal_66"
                      >
                        <div
                          className="modal-content"
                        >
                          <h4>
                            Maximum: 
                          </h4>
                        </div>
                        <div
                          className="modal-footer"
                        >
                          <Button
                            flat={true}
                            key=".0"
                            modal="close"
                            node="button"
                            waves="light"
                          >
                            <button
                              className="btn waves-effect waves-light btn-flat modal-action modal-close"
                              disabled={false}
                            >
                              Close
                            </button>
                          </Button>
                        </div>
                      </div>
                    </Portal>
                  </div>
                </Modal>
              </ul>
            </span>
          </Dropdown>
          <Input
            checked={false}
            className="range"
            label="Enter Range 1 <= 5000"
            onChange={[Function]}
            s={2}
            type="text"
          >
            <div
              className="col input-field s2"
            >
              <input
                checked={false}
                className="range"
                id="input_67"
                onChange={[Function]}
                type="text"
              />
              <label
                className=""
                htmlFor="input_67"
              >
                Enter Range 1 &lt;= 5000
              </label>
            </div>
          </Input>
          <Button
            className="generate"
            node="button"
            onClick={[Function]}
            s={6}
            style={
              Object {
                "marginLeft": 40,
                "marginTop": 20,
              }
            }
            waves="light"
          >
            <button
              className="btn waves-effect waves-light generate"
              disabled={false}
              onClick={[Function]}
              s={6}
              style={
                Object {
                  "marginLeft": 40,
                  "marginTop": 20,
                }
              }
            >
              GENERATE
            </button>
          </Button>
        </div>
      </Row>
    </div>
  </div>
</App>
`;
